### DISCUSSION TITLE
calculator script

### DISCUSSION THREAD

**Discussion by divya-gh** [avinashkranjan/Amazing-Python-Scripts#447] [Q&A] [answered]
I had a question regarding the eval() function in the calculator script. I see that when the previous value is not 0, its value has been assigned with a new calculation result as seen below
![eval_operation](https://user-images.githubusercontent.com/74730673/110074124-4c868880-7d46-11eb-8870-de3649893485.jpg)

My question is, Why are we trying to assign a value to the previous variable? 

Challenges: 
It was hard to understand what was expected. more documentation needed with expected results.

**Comment by avinashkranjan** [comment, marked as answer]
> I had a question regarding the eval() function in the calculator script. I see that when the previous value is not 0, its value has been assigned with a new calculation result as seen below
> ![eval_operation](https://user-images.githubusercontent.com/74730673/110074124-4c868880-7d46-11eb-8870-de3649893485.jpg)
> 
> My question is, Why are we trying to assign a value to the previous variable?
> 
> Challenges:
> It was hard to understand what was expected. more documentation needed with expected results.

So That you can continue to do the calculation, In case of a complex calculation or you need to calculate something. So, It assigns the new answer to the variable for convenience else it would not be convenient if you need to remember last answer to calculate next expression.
